### PR TITLE
fix(1007): adding shared keyring mode to type unit

### LINKED ISSUE #1007: Importing keys to _ima keyring in pre-pivot/integrity module does not work due to systemd restricting keyrings
dracut's integrity module tries to load the public key inside initrd, but receives -ACCES:

```
dracut-pre-pivot[554]: /lib/dracut-lib.sh@431(source_all): . //lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh
dracut-pre-pivot[554]: ///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@3(source): SECURITYFSDIR=/sys/kernel/security
dracut-pre-pivot[554]: ///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@4(source): IMASECDIR=/sys/kernel/security/ima
dracut-pre-pivot[554]: ///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@5(source): IMACONFIG=/sysroot/etc/sysconfig/ima
dracut-pre-pivot[554]: ///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@43(source): '[' '!' -e /sys/kernel/security/ima ']'
////lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@51(source): keyctl describe %keyring:.ima
Can't find 'keyring:.ima'
///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@51(source): line=
///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@52(source): '[' 1 -eq 0 ']'
////lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@55(source): keyctl search @u keyring _ima
keyctl_search: Required key not available
///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@55(source): _ima_id=
///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@56(source): '[' -z '' ']'
////lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@57(source): keyctl newring _ima @u
///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@57(source): _ima_id=541363765
///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@62(source): load_x509_keys 541363765
///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@9(load_x509_keys): KEYRING_ID=541363765
///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@12(load_x509_keys): '[' -f /sysroot/etc/sysconfig/ima ']'
///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@16(load_x509_keys): '[' -z '' ']'
///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@17(load_x509_keys): IMAKEYSDIR=/etc/keys/ima
////lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@20(load_x509_keys): ls /sysroot/etc/keys/ima/x509_evm.der
///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@20(load_x509_keys): PUBKEY_LIST=/sysroot/etc/keys/ima/x509_evm.der
///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@21(load_x509_keys): for PUBKEY in ${PUBKEY_LIST}
///lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@23(load_x509_keys): '[' '!' -f /sysroot/etc/keys/ima/x509_evm.der ']'
////lib/dracut/hooks/pre-pivot/61-ima-keys-load.sh@30(load_x509_keys): evmctl import /sysroot/etc/keys/ima/x509_evm.der 541363765
Reading to /sysroot/etc/keys/ima/x509_evm.der
Importing public key 5ac982c4 from file /sysroot/etc/keys/ima/x509_evm.der into keyring 541363765
add_key failed
errno: Permission denied (13)
```

strace of this inside initrd:
```
575   write(2, "Importing public key 5ac982c4 fr"..., 98) = 98
575   add_key("asymmetric", NULL, "0\202\2\2040\202\1\356\240\3\2\1\2\2\t\0\242\5\2726G\242\331\3220\f\6\10*\205\3\7"..., 648, 131199347) = -1 EACCES (Permission denied)
575   write(2, "add_key failed\n", 15)  = 15
575   write(2, "errno: Permission denied (13)\n", 30) = 30
```

Adding `KeyringMode=shared` to `/usr/lib/dracut/modules.d/98dracut-systemd/dracut-pre-pivot.service` and rebuilding initrd makes `evmctl import` work.

https://github.com/systemd/systemd/issues/5522 helped me to come to such a workaround.

For now I am not sure why this helped and how to fix the problem properly.

### PR DESCRIPTION
**Pull request by johannbg**
This pull request adds a shared keyring mode to dracut-pre-pivot.service as requested thus fixes #1007

### CODE REVIEW

File: modules.d/98dracut-systemd/dracut-pre-pivot.service [hunk 1]
```diff
@@ -30,6 +30,7 @@ StandardInput=null
 StandardError=journal+console
 KillMode=process
 RemainAfterExit=yes
+KeyringMode=shared
```
Comment (mikhailnov) [inline modules.d/98dracut-systemd/dracut-pre-pivot.service:33]: There is a trailing whitespace in the end of this string

**Review by mikhailnov — CHANGES_REQUESTED** [review]

**Comment by mikhailnov** [comment]
Now there are two similar commits here, could you please make one and force push?

```
git reset --soft <ID of the first commit>
git add .
git commit --amend
git push origin +<branch-name>
```

**Comment by mikhailnov** [comment]
Also, you can skip a line after the first line of the commit message and add text like this:

`Fixes: https://github.com/dracutdevs/dracut/issues/1007`

then github will automatically make a link from the PR to this issue and from the issue to the PR and will automatically close the ussie when the PR is merged. And there will be a link to the issue in the history of commits.

**Comment by johannbg** [comment]
@mikhailnov thanks for the tips. :+1:

**Comment by mikhailnov** [comment]
Now it looks much better, thanks!

**Review by haraldh — APPROVED** [review]

**Comment by johannbg** [comment]
@danimo  ping...

**Review by danimo — APPROVED** [review]

**Pull request merged**